sklearn/ensemble/_stacking.py

```diff
@@ -392,8 +400,43 @@ def _sk_visual_block_with_final_estimator(self, final_estimator):
         )
         return _VisualBlock("serial", (parallel, final_block), dash_wrapped=False)
 
+    def get_metadata_routing(self):
+        """Get metadata routing of this object.
+
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        .. versionadded:: 1.5
+
+        Returns
+        -------
+        routing : MetadataRouter
+            A :class:`~sklearn.utils.metadata_routing.MetadataRouter` encapsulating
+            routing information.
+        """
+        router = MetadataRouter(owner=self.__class__.__name__)
+
+        # `self.estimators` is a list of (name, est) tuples
+        for name, estimator in self.estimators:
+            router.add(
+                **{name: estimator},
+                method_mapping=MethodMapping().add(callee="fit", caller="fit"),
+            )
+
+        try:
+            final_estimator_ = self.final_estimator_
+        except AttributeError:
+            final_estimator_ = self.final_estimator
```

Comment:
Even though this seems okay, can we do it in the following manner which avoids catching errors
```suggestion
        final_estimator_ = self.final_estimator_ if hasattr(self, final_estimator_) else self. self.final_estimator
```

Reply:
That shows how uncertain we are on this pattern 😅

Reply:
I don't have a passionate opinion about this. I will leave this as is, unless someone really wants to change that.